exporter/signalfxexporter/config.go

```diff
@@ -160,8 +157,11 @@ func (cfg *Config) validateConfig() error {
 		return errors.New(`cannot have a negative "timeout"`)
 	}
 
-	if cfg.MaxConnections < 0 {
-		return errors.New(`cannot have a negative "max_connections"`)
+	if cfg.MaxConnections != nil {
+		logger.Warn(`"max_connections" is deprecated, use "max_idle_conns" or "max_idle_conns_per_host" instead`)
```

Comment:
Realized we should probably have a check for the new values not being nil and log that only they will be used.

Reply:
Pushed the changes accordingly